src/Beacon.jsx

```diff
@@ -157,6 +164,23 @@ export class Beacon extends Component {
         this.setState({ targetElement, targetBounds }); // eslint-disable-line react/no-did-update-set-state
       }
     }
+
+    if (this.state.tooltip && !this.state.arrowPosition) {
+      // Calculate relative arrow position when ref is available
+      setTimeout(() => {
+        const arrowStyleLeft = this.getArrowHorizontalPosition(
+          targetElement.getBoundingClientRect(),
+          this.refs['tooltip'].getBoundingClientRect(),
```

Comment:
I would suggest to set refs with callbacks instead of strings anyway:

```jsx
class Foo extends React.Component (
  contructor(props) {
   super(props);
  // ...
  }

 // ...

  setBar(node){
    this.bar = node;
  }

  render(){
    return (
      <div>
        <span ref={this.setBar}>Example...</span>
      </div>
    );
  }
}
```
What official doc says: https://facebook.github.io/react/docs/more-about-refs.html
> Note:
> Although string refs are not deprecated, they are considered legacy, and will likely be deprecated at some point in the future. Callback refs are preferred.